src/runner.rs

```diff
@@ -121,8 +123,15 @@ fn parse_test(test_path: &Path, config: &TestConfig) -> InnerTestResult<Test> {
                 expected_exit_status = Some(status.parse().map_err(|err| {
                     InnerTestError::ErrorParsingExitStatus(test_path.to_owned(), status.to_owned(), err)
                 })?);
+            } else {
+                append_line(&mut rest, line);
             }
         } else {
+            // Both expected_stdout and expected_stderr need a blank line at the end,
+            // strip that out.
```

Comment:
The parsing is a bit tricky here, I for one didn't initially read the parsing here correctly. Consider this:
```
# expected stdout:
# a
# expected stderr:
# b
```
(Assuming python sane settings when running the program)

We then get the following error:
```
examples/thingy.py: Actual stdout differs from expected stdout:
   | a
   | expected stderr:
   | b
```
The state only changes when a blank line is hit or we reach EOF. Ofcourse, this is the behavior one wants since the output could contain the string "expected stderr:". I tried to clarify this in the comment but I might have failed.

Reply:
That is correct and expected yes. Perhaps something for me to be clearer about in the docs as well. I originally read your comment wrong then, thanks for the rewording + clarification.